### PR TITLE
Feature/retreive master data

### PR DESCRIPTION
**Pull request by rahultandonLnG**
Added Services for retreiving master data and populated master data on add stock UI.

### CODE REVIEW

File: Pspcl/Pspcl.Services/Interfaces/IStockService.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+﻿using Pspcl.Core.Domain;
+
+namespace Pspcl.Services.Interfaces
+{
+	public interface IStockService
+	{
+		List<MaterialGroup> GetAllMaterialGroups(bool? onlyActive = false);
+		List<MaterialType> GetAllMaterialTypes(int? materialGroupId, bool? showHidden=false);
```
Comment (sandeepkumarlng) [inline Pspcl/Pspcl.Services/Interfaces/IStockService.cs]: Change show hidden to onlyActive

File: Pspcl/Pspcl.Services/StockService.cs [hunk 2]
```diff
@@ -1,20 +1,89 @@
...
+        private readonly ApplicationDbContext _dbcontext;
+        public StockService(ApplicationDbContext dbContext)
         {
-            _dbcontext = context;
-                        
+            _dbcontext = dbContext;
         }
 
-        
+        public List<MaterialGroup> GetAllMaterialGroups(bool? onlyActive = false)
```
Comment (sandeepkumarlng) [inline Pspcl/Pspcl.Services/StockService.cs:17]: I did an example for you, is this what I told you to do to filter records ?

File: Pspcl/Pspcl.Services/StockService.cs [hunk 3]
```diff
@@ -1,20 +1,89 @@
...
+        {
+            if (!onlyActive.HasValue)
+            {
+                return _dbcontext.MaterialGroup.ToList();
+            }
+            else
+            {
+                if (onlyActive.HasValue)
```
Comment (sandeepkumarlng) [inline Pspcl/Pspcl.Services/StockService.cs]: Do you need these if else checks ?

File: Pspcl/Pspcl.Web/Controllers/StockViewController.cs [hunk 4]
```diff
@@ -1,27 +1,42 @@
...
-        {
-         
-            return View("~/Views/StockView/AddStock.cshtml");
-        }
+        public StockViewController(IStockService addStockService)
+		{
+			_StockService = addStockService;
+		}
 
+		public IActionResult StockView()
```
Comment (sandeepkumarlng) [inline Pspcl/Pspcl.Web/Controllers/StockViewController.cs]: Why name is stock view ?

Comment (rahultandonLnG) [reply Pspcl/Pspcl.Web/Controllers/StockViewController.cs:18]: Abhijeet changed this and it was changed when I took pull from develop.

**Review by sandeepkumarlng — APPROVED** [review]

**Pull request merged**